src/API/Site/Controllers/Ads/AccountController.php

```diff
@@ -27,45 +19,22 @@
  *
  * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Ads
  */
-class AccountController extends BaseOptionsController {
+class AccountController extends BaseController {
 
 	/**
-	 * @var ContainerInterface
+	 * @var AccountService
 	 */
-	protected $container;
-
-	/**
-	 * @var Middleware
-	 */
-	protected $middleware;
-
-	/**
-	 * @var AdsAccountState
-	 */
-	protected $account_state;
-
-	/**
-	 * @var Ads
-	 */
-	protected $ads;
-
-	/**
-	 * @var AdsConversionAction
-	 */
-	protected $ads_conversion_action;
+	protected $account;
 
 	/**
 	 * AccountController constructor.
 	 *
-	 * @param ContainerInterface $container
+	 * @param RESTServer     $server
+	 * @param AccountService $account
```

Comment:
Nice suggestion, although I think it's a little more than a container, since it has a account state handler within that class. I added the following comment instead:

> Service used to access / update Ads account data.